libs/gl-client/src/signer/resolve.rs

```diff
@@ -0,0 +1,103 @@
+    /// the contents of the context request. TODOs in here may
+    /// indicate ways to strengthen the verification.
+    pub fn try_resolve(req: &Message, reqctx: &Vec<Request>) -> Result<(), Error> {
+        log::trace!("Resolving {:?}", req);
+        // Some requests do not need a justification. For example we
+        // reconnect automatically, so there may not even be a context
+        // request pending which would skip the entire stack below, so
+        // we do an early pass:
+        let accept = match req {
+            // Commands that simply have no context to check against
+            Message::Ecdh(_) => true,
+            Message::Ping(_) => true,
+            Message::Pong(_) => true,
+            Message::SignChannelAnnouncement(_) => true,
+            Message::SignChannelUpdate(_) => true,
+            Message::SignNodeAnnouncement(_) => true,
+            // Duplicate verification with VLS, we defer to VLS
+            Message::GetChannelBasepoints(_) => true,
+            Message::ValidateCommitmentTx(_) => true,
+            Message::SignWithdrawal(_) => true,
+            Message::ReadyChannel(_) => true,
+            Message::GetPerCommitmentPoint(_) => true,
+            Message::ValidateRevocation(_) => true,
+            Message::NewChannel(_) => true,
+	    Message::SignCommitmentTx(_) => true,
+	    Message::SignGossipMessage(_) => true,
+	    Message::SignMutualCloseTx(_) => true,
+	    Message::SignMutualCloseTx2(_) => true,
+	    Message::SignRemoteCommitmentTx(_) => true,
+	    Message::SignRemoteCommitmentTx2(_) => true,
+	    // Resolution of an existing HTLC, we should never not try
+	    // to grab funds if we can.
+	    Message::SignPenaltyToUs(_) => true,
+	    Message::SignAnyPenaltyToUs(_) => true,
+	    Message::SignAnyDelayedPaymentToUs(_) => true,
+	    Message::SignAnyLocalHtlcTx(_) => true,
+	    Message::SignAnyRemoteHtlcToUs(_) => true,
+	    // Default to rejecting, punting the decision to the next
+	    // step.
+            _ => false,
+        };
+
+        // If we found a resolution, then there is no point in trying
+        // to match up further.
+        if accept {
+            log::trace!(
+                "Request {:?} resolved with no context request required",
+                req
+            );
+            return Ok(());
+        }
+
+        for cr in reqctx {
+            let accept = match (dbg!(req), dbg!(cr)) {
+                (Message::SignMessage(m1), Request::SignMessage(m2)) => {
+                    m1.message.0.clone() == m2.message.as_bytes()
+                }
+                (Message::NewChannel(m1), Request::FundChannel(m2)) => {
+                    // Different node_id? Reject!
+                    m1.node_id.0 == m2.id.as_slice()
+                    // TODO: Add `close_to` to allowlist for the close
+                    // later on
+                }
+                (Message::NewChannel(m1), Request::GlFundChannel(m2)) => {
```

Comment:
may be good to also match `Message::ReadyChannel` so we can validate the amount and shutdown script

Reply:
also, when does the client generate a `GlFundChannel` and when a `FundChannel`?

Reply:
Good catch, I will add a match-arm for `ReadyChannel`.

The difference between `GlXyz` and `Xyz` is that one is the legacy `greenlight.proto` variant, and the non-prefixed one is the modern cln-grpc `node.proto` variant. We will be retiring the `greenlight.proto` version in v0.2 soon :tm: